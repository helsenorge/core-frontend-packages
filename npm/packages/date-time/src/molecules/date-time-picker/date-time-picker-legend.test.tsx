import * as React from 'react';
import { mount } from 'enzyme';

import DateTimePickerLegend from './date-time-picker-legend';

describe('DateTimePicker date-time-picker-legend', () => {
  describe('Gitt at DateTimePickerLegend skal rendres', () => {
    describe('Når den bare har default verdier', () => {
      const wrapper = mount(<DateTimePickerLegend legend={'myLabel'} isRequired={false} />);
      it('Så rendres den med rikitg id legend tekst', () => {
        expect(wrapper.find('legend').text()).toEqual('myLabel');
        expect(wrapper.render()).toMatchSnapshot();
      });
    });

    describe('Når den er required og har requiredLabel', () => {
      const wrapper = mount(<DateTimePickerLegend legend={'myLabel'} isRequired={true} requiredLabel={'MyRequiredLabel'} />);
      it('Så rendres den med rikitg legend (legend med requiredLabel og eksplisitt format)', () => {
        expect(wrapper.find('legend').text()).toEqual('myLabel MyRequiredLabel');
        expect(wrapper.render()).toMatchSnapshot();
      });
    });

    describe('Når den har optionalLabel', () => {
      const wrapper = mount(<DateTimePickerLegend legend={'myLabel'} isRequired={false} optionalLabel={'MyOptionalLabel'} />);
      it('Så rendres den med rikitg legend (legend med optionalLabel og eksplisitt format)', () => {
        expect(wrapper.find('legend').text()).toEqual('myLabel MyOptionalLabel');
        expect(wrapper.render()).toMatchSnapshot();
      });
    });

    describe('Når den har helpButton', () => {
      const wrapper = mount(
        <DateTimePickerLegend legend={'myLabel'} isRequired={false} helpButton={<span className={'helpButton'}>{'helpButton'}</span>} />
      );
      it('Så rendres den med helpButton nederst i legenden', () => {
        expect(wrapper.find('legend').find('.helpButton').length).toEqual(1);
        expect(wrapper.render()).toMatchSnapshot();
      });
    });

    describe('Når den har subLabel', () => {
      const wrapper = mount(
        <DateTimePickerLegend legend={'myLabel'} isRequired={false} subLabel={<span className={'subLabel'}>{'subLabel'}</span>} />
      );
      it('Så rendres den med subLabel nederst i legenden', () => {
        expect(wrapper.find('.subLabel').length).toEqual(1);
        expect(wrapper.render()).toMatchSnapshot();
      });
    });
  });
});
