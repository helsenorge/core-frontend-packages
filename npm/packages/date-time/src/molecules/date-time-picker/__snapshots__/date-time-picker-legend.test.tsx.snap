// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateTimePicker date-time-picker-legend Gitt at DateTimePickerLegend skal rendres Når den bare har default verdier Så rendres den med rikitg id legend tekst 1`] = `
<legend>
  myLabel
</legend>
`;

exports[`DateTimePicker date-time-picker-legend Gitt at DateTimePickerLegend skal rendres Når den er required og har requiredLabel Så rendres den med rikitg legend (legend med requiredLabel og eksplisitt format) 1`] = `
<legend>
  myLabel
  <em>
     MyRequiredLabel
  </em>
</legend>
`;

exports[`DateTimePicker date-time-picker-legend Gitt at DateTimePickerLegend skal rendres Når den har helpButton Så rendres den med helpButton nederst i legenden 1`] = `
<legend>
  myLabel
  <span
    class="helpButton"
  >
    helpButton
  </span>
</legend>
`;

exports[`DateTimePicker date-time-picker-legend Gitt at DateTimePickerLegend skal rendres Når den har optionalLabel Så rendres den med rikitg legend (legend med optionalLabel og eksplisitt format) 1`] = `
<legend>
  myLabel
  <em>
     MyOptionalLabel
  </em>
</legend>
`;

exports[`DateTimePicker date-time-picker-legend Gitt at DateTimePickerLegend skal rendres Når den har subLabel Så rendres den med subLabel nederst i legenden 1`] = `
<legend>
  myLabel
  <span
    class="label__sublabel"
  >
    <span
      class="subLabel"
    >
      subLabel
    </span>
  </span>
</legend>
`;
